include/maths/CBoostedTreeImpl.h

```diff
@@ -402,17 +412,29 @@ class MATHS_EXPORT CBoostedTreeImpl final {
         //! Get the row mask for this leaf node.
         core::CPackedBitVector& rowMask() { return m_RowMask; }
 
+        //! Get the memory used by this object.
+        std::size_t memoryUsage() const {
+            std::size_t mem{core::CMemory::dynamicSize(m_FeatureBag)};
+            mem += core::CMemory::dynamicSize(m_RowMask);
+            mem += core::CMemory::dynamicSize(m_Gradients);
+            mem += core::CMemory::dynamicSize(m_Curvatures);
+            mem += core::CMemory::dynamicSize(m_MissingGradients);
+            mem += core::CMemory::dynamicSize(m_MissingCurvatures);
+            return mem;
+        }
+
         //! Estimate maximum leaf statistics bookkeeping memory for training
         //! boosted trees on data frames with \p numberRows rows, \p numberCols columns
         //! with specified settings for \p featureBagFraction and \p numberSplitsPerFeature
-        static size_t estimateMemoryUsage(std::size_t numberRows,
-                                          std::size_t numberCols,
-                                          double featureBagFraction,
-                                          std::size_t numberSplitsPerFeature) {
+        static std::size_t estimateMemoryUsage(std::size_t numberRows,
+                                               std::size_t numberCols,
+                                               double featureBagFraction,
+                                               std::size_t numberSplitsPerFeature) {
             std::size_t featureBagSize{
-                static_cast<std::size_t>(ceil(featureBagFraction) * (numberCols - 1)) *
+                static_cast<std::size_t>(std::ceil(
+                    featureBagFraction * static_cast<double>(numberCols - 1))) *
                 sizeof(std::size_t)};
-            std::size_t rowMaskSize{numberRows / 32};
+            std::size_t rowMaskSize{numberRows / 256};
```

Comment:
Could we document this magic number?

Reply:
See [68158b7](https://github.com/elastic/ml-cpp/pull/607/commits/68158b7f586e39cdd6e86b205ea9efab69b02d7e).